File: 백준/Silver/1065. 한수/한수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());

        if (N < 100) {
            System.out.println(N);
        } else {
            int cnt = 99;
            for (int num = 100; num <= N; num++) {
                if (isSequence(num)) {
                    cnt++;
                }
            }
            System.out.println(cnt);
        }
    }

    static boolean isSequence(int num) {
        int pre = num % 10;
        num /= 10;
        int next = num % 10;
        int diff = pre - next;
        while (true) {
            pre = next;
            num /= 10;
            next = num % 10;
            if (num == 0) break;
            if (diff != pre - next) {
                return false;
            }
        }
        return true;
    }
}

File: 백준/Silver/1065. 한수/한수.js
let input = require('fs').readFileSync('/dev/stdin').toString().trim();
let num = parseInt(input);
let cnt = 99;

// num이 99 이하일 경우 바로 출력
if (num <= 99) {
    console.log(num);
} else {
    for (let i = 100; i <= num; i++) {
        if (isEqualDiff(i)) {
            cnt++;
        }
    }
    console.log(cnt);
}

function isEqualDiff(num) {
    let str = num.toString(); // 숫자를 문자열로 변환
    let diff = str[1] - str[0]; // 첫 두 자리의 차이 계산

    for (let j = 2; j < str.length; j++) {
        if (str[j] - str[j - 1] != diff) {
            return false; // 등차 수열이 아닌 경우
        }
    }
    return true; // 등차 수열일 경우
}